packages/date-picker/src/vaadin-date-picker.js

```diff
@@ -238,7 +230,7 @@ class DatePicker extends DatePickerMixin(InputControlMixin(ThemableMixin(Element
   /** @private */
   _toggle(e) {
     e.stopPropagation();
-    this[this._overlayInitialized && this.$.overlay.opened ? 'close' : 'open']();
+    this[this.$.overlay.opened ? 'close' : 'open']();
```

Comment:
```suggestion
if (this.$.overlay.opened) {
  this.close();
else {
  this.open();
}
```
Nit: Not a fan of this dynamic stuff personally, also hurts with code navigation and search for references.